### PR TITLE
test: rewrite snapshots without unicode prefixes

### PR DESCRIPTION
**Pull request by joshuarli**
Under python 2, repr will literally add the unicode prefix in.

```
>>> "%r" % u"aa"
"u'aa'"
```

To make these snapshots compatible with testing under py3, I'm removing uses of repr and regenerating these snapshots. I wasn't able to find a way to get py3 repr to insert literal u? If there is a way without being hacky, I'd love to know it, because then that would obviate the need for this huge diff.

### CODE REVIEW

File: tests/sentry/grouping/test_variants.py [hunk 1]
```diff
@@ -44,7 +45,7 @@ def _dump_component(component, indent):
             # We do not want to dump the config
             continue
         else:
-            lines.append("%s%s: %r" % ("  " * indent, key, value))
+            lines.append("%s%s: %s" % ("  " * indent, key, value))
```
Comment (untitaker) [inline tests/sentry/grouping/test_variants.py]: Can you do `repr()` on Python 2 and `ascii()` on Python 3 such that we can differentiate between `"123"` and `123`? And remove a leading `u"` manually.

**Review by untitaker — CHANGES_REQUESTED** [review]

File: tests/sentry/grouping/test_variants.py (hunk 1 above)
Comment (joshuarli) [reply tests/sentry/grouping/test_variants.py]: > And remove a leading u" manually.

Like, literally strip leading `u"` and last `"` from the line?

Comment (untitaker) [reply tests/sentry/grouping/test_variants.py]: Basically the concern is that we cannot differentiate between strings that are valid integers and genuine integers in the current snapshots as you rewrote them.

@mitsuhiko curious why didn't we just json/yaml dump here?

Comment (joshuarli) [reply tests/sentry/grouping/test_variants.py]: @untitaker I dumped yaml, it looks a lot more verbose and less readable than the custom logic that's currently in place:

image

```diff
diff --git a/tests/sentry/grouping/test_variants.py b/tests/sentry/grouping/test_variants.py
index 01ca808187..d41a1f659c 100644
--- a/tests/sentry/grouping/test_variants.py
+++ b/tests/sentry/grouping/test_variants.py
@@ -100,12 +100,15 @@ def test_event_hash_variant(insta_snapshot, config_name, test_name, log):
     # break stuff later on.
     evt.project = None

+    import yaml
     rv = []
     for (key, value) in sorted(evt.get_grouping_variants().items()):
         if rv:
             rv.append("-" * 74)
         rv.append("%s:" % key)
-        dump_variant(value, rv, 1)
+        data = value.as_dict()
+        data.pop("config", None)
+        rv.append(yaml.dump(data))
     output = "\n".join(rv)
```

Comment (untitaker) [reply tests/sentry/grouping/test_variants.py]: okay, just make sure that you can tell integers and strings apart. Might even be fine to `json.dumps`

Comment (joshuarli) [reply tests/sentry/grouping/test_variants.py]: That is a great idea!! 

```
>>> repr(2)
'2'
>>> repr(u"2")
"u'2'"
>>> json.dumps(2)
'2'
>>> json.dumps(u"2")
'"2"'
```

Comment (joshuarli) [reply tests/sentry/grouping/test_variants.py]: @untitaker Confirmed py3 likes the py2 generated snapshots with json.dumps in https://github.com/getsentry/sentry/pull/17288. :)

**Review by untitaker — APPROVED** [review]
good luck

**Comment by joshuarli** [comment]
thank

**Pull request merged**